src/math/p5.Vector.js

```diff
@@ -584,22 +602,31 @@ p5.Vector.prototype.sub = function sub(x, y, z) {
  * </code>
  * </div>
  */
-p5.Vector.prototype.mult = function mult(n) {
-  if (!(typeof n === 'number' && isFinite(n))) {
-    console.warn(
-      'p5.Vector.prototype.mult:',
-      'n is undefined or not a finite number'
-    );
+p5.Vector.prototype.mult = function mult(x, y, z) {
+  if (x instanceof p5.Vector) {
+    this.x *= x.x || 0;
+    this.y *= x.y || 0;
+    this.z *= x.z || 0;
     return this;
   }
-  this.x *= n;
-  this.y *= n;
-  this.z *= n;
+  if (x instanceof Array) {
+    this.x *= x[0] || 0;
+    this.y *= x[1] || 0;
+    this.z *= x[2] || 0;
+    return this;
+  }
+  const _x = x || 1;
+  const _y = y || x;
+  const _z = z || x;
```

Comment:
If it defaults to `x` in case `y` or `z` is not defined while `x` is also undefined, `x` will be `1` while `y` and `z` will be undefined. Also I'm not sure if `x` should default to `1` here if falsy, same reason as above, it makes debugging harder.

Reply:
Woops, I think that 1 is supposed to be a 0. 

Maybe I'm not understanding your comment but I think in the case of `x` being defined, and `y` and `z` being undefined, wouldn't they all end up as `x`? I wrote it this way to keep the functionality of being able to provide a single scalar value to multiply against all the vector components. 

```
const val = 2;
const _x =  val || 1; // _x equals 2
const _y = y || val; // y is undefined so _y also equals 2
const _z = z || val; // z is undefined so _z also equals 2
```

One tricky case that I just thought of is what to do in the case that someone provides just both x and y values but doesn't provide a z. 

This also makes me wonder if we should also allow `.add` and `.sub` to be used with a single scalar value like we already did for `.mult and .div`

Reply:
My explanation is a bit confusing (it's confusing me as well) so I'll just illustrate in code:

```javascript
const val = undefined; // Not valid but just to illustrate
const _x =  val || 1; // _x equals 1
const _y = y || val; // y is undefined so _y equals val which equals to undefined
const _z = z || val; // z is undefined so _z equals val which equals to undefined
```

Rather should probably be

```javascript
const val = undefined; // Not valid but just to illustrate
const _x =  val || 1; // _x equals 1
const _y = y || _x; // y is undefined so _y equals _x which equals to 1
const _z = z || _x; // z is undefined so _z equals _x which equals to 1
```

And I think default to `1` is actually correct here as if `mult()` is not called with any arguments or called with an undefined argument, the vector should not be modified and in this case is to multiply by 1.

For the case of users providing only `x` and `y` but not `z`, it is technically an undocumented/invalid signature which we don't have to account for. `add()` and `sub()` probably should not be modified to add scalar across all values as that would be a breaking change.

Reply:
With type checking, we no longer have this issue of the defaults, since you'll just get back an unchanged vector if something was undefined / nan